RetailCoder.VBE/Navigation/CodeExplorer/CodeExplorerComponentViewModel.cs

```diff
@@ -47,7 +47,53 @@ public CodeExplorerComponentViewModel(CodeExplorerItemViewModel parent, Declarat
                                 .OrderBy(item => item.QualifiedSelection.Selection.StartLine)
                                 .Select(item => new CodeExplorerMemberViewModel(this, item, grouping)))
                                 .ToList<CodeExplorerItemViewModel>();
-            
+
+            _name = _declaration.IdentifierName;
+
+            var component = declaration.QualifiedName.QualifiedModuleName.Component;
+            if (component.Type == vbext_ComponentType.vbext_ct_Document)
+            {
+                try
+                {
+                    var parenthesizedName = component.Properties.Item("Name").Value.ToString();
+
+                    if (ContainsBuiltinDocumentPropertiesProperty())
+                    {
+                        CodeExplorerItemViewModel node = this;
+                        while (node.Parent != null)
+                        {
+                            node = node.Parent;
+                        }
+
+                        ((CodeExplorerProjectViewModel) node).SetParenthesizedName(parenthesizedName);
+                    }
+                    else
+                    {
+                        _name += " (" + parenthesizedName + ")";
+                    }
+                }
+                catch
+                {
+                    // gotcha! (this means that the property either doesn't exist or we weren't able to get it for some reason)
```

Comment:
This will throw a COMException if the `Name` property doesn't exist:

     var nameProperty = component.Properties.Item("Name");

This shouldn't:

    var nameProperty = component.Properties.Cast<Property>().SingleOrDefault(p => p.Name = "Name");

And then go on if `nameProperty` isn't `null`.